module.nix

```
@@ -89,8 +136,27 @@ in
           requires = [
             "async-nix-post-build-hook.socket"
           ];
+          script = ''
+            set -euo pipefail
+            shopt -u nullglob
+            ls -l ''${CREDENTIALS_DIRECTORY}
+            # Load all credentials into env if they are in UPPER_SNAKE form.
+            if [[ -n "''${CREDENTIALS_DIRECTORY:-}" ]]; then
+              for file in "$CREDENTIALS_DIRECTORY"/*; do
+                key=$(basename "$file")
+                if [[ $key =~ ^[A-Z0-9_]+$ ]]; then
+                  echo "Environ $key"
+                  export "$key=$(< "$file")"
+                fi
+              done
+            fi
+            exec ${cfg.package}/bin/queued-build-hook daemon --hook ${hook} --retry-interval ${toString cfg.retryInterval} --retry-interval ${toString cfg.retries} --concurrency ${toString cfg.concurrency} 
+          '';
           serviceConfig = {
-            ExecStart = "${cfg.package}/bin/queued-build-hook daemon --hook ${hook}";
+            DynamicUser = true;
```

Not strictly needed but looks nicer in a process list. systemd provides a PAM module to resolve those dynamic users/groups.

```suggestion
            DynamicUser = true;
            User = "queued-build-hook";
            Group = "queued-build-hook";
```